content/api/commands/attachfile.md

```diff
@@ -0,0 +1,235 @@
+
+### From a file on disk
+
+```javascript
+cy.get('input[type=file]').attachFile('file.json')
+cy.get('input[type=file]').attachFile('file.png')
+```
+
+If given a path, `.attachFile()` will attach the file exactly as it exists on disk. This is the preferred way to work with files on disk, avoiding many encoding-related pitfalls.
+
+### On a hidden input
+
+```javascript
+cy.get('input[type=file]').attachFile('file.json', { force: true })
+```
+
+In many cases in modern applications, the underlying file input is hidden from view, and activated by a user clicking on a button. In these cases, you will need to tell Cypress to ignore its actionability checks and attach the file even though a user would not be able to directly activate the file input.
+
+### From a fixture
+```javascript
+cy.fixture('file.json', { encoding: null }).as('myFixture')
+cy.get('input[type=file]').attachFile('@myFixture')
+```
+
+Note the use of `null` encoding. By default, `cy.fixture()` and `cy.readFile()` attempt to interpret files read from disk, which would result in a JSON file being decoded and re-encoded as a utf-8 string - the contents would be preserved, but formatting would not be and the encoding might change. See [`cy.fixture`](/api/commands/fixture) or [`cy.readFile`](/api/commands/readfile) for more details on file encoding.
+
+### From an API response
+
+```javascript
+cy.request('http://localhost:8888/users/827')
+  .its('body')
+  .as('responseBody')
+
+cy.get('input[type=file]').attachFile('@responseBody')
+```
+
+### Processing data inside the test
+
+```javascript
+cy.readFile('users.json').then((users => {
+  users[0].username = 'JohnCena'
+})).as('myFile')
+
+cy.get('input[type=file]').attachFile('@myFile')
+```
+
+### Attaching multiple files
+
+```javascript
+cy.get('input[type=file]').attachFile([
+  'file1.json',
+  'file2.json',
+  'file3.json',
+])
+```
+
+<Alert type="warning">
+
+This will fail unless the file input has the `multiple` property.
+
+</Alert>
+
+### Attaching a file with custom filename and lastModified
+
+```javascript
+cy.get('input[type=file]').attachFile({
+  contents: 'path/to/file.json',
+  fileName: 'users.json',
+  lastModified: new Date('Feb 18 1989').valueOf()
+})
+```
+
+### Dropping a file on the document
+
+```javascript
+cy.document().attachFile('file.json', { action: 'drag-n-drop' })
+```
+
+## Notes
+
+### Existence
+
+#### Default file existence assertion
+
+When passed a file path, `.attachFile()` asserts that the file exists and will fail if it does not exist. It will retry reading the file if it does not initially exist until the file exists or the command times out.
+
+```javascript
+// will fail after the defaultCommandTimeout is reached
+cy.get('input[type=file]').attachFile('does-not-exist.yaml')
+```
+
+### Actionability
+
+#### The element must first reach actionability
+
+`.attachFile()` is an "action command" that follows all the rules
+[defined here](/guides/core-concepts/interacting-with-elements).
```

Comment:
```suggestion
of [Actionability](/guides/core-concepts/interacting-with-elements#Actionability).
```

Reply:
I copied this directly from another command, and it seems to be the common pattern in our docs.

![image](https://user-images.githubusercontent.com/3003404/146038924-2b69cf3b-a210-445b-a2aa-105b6ec5b90e.png)

Do you want to change this everywhere?

Reply:
I think we should change it everywhere, but that can also be done in a separate, follow-up PR instead of further complicating this one. If we do that, just leave it as-is. Your call!

Reply:
It's a good change but a small one, so I went ahead an applied it to all the docs here.